.github/workflows/deploy.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy Beta
+
+on:
+  push:
+    branches: [site]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build and Deploy
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.0.0
+
+      - name: Setup Node
+        uses: actions/setup-node@v1.1.0
+        with:
+          node-version: lts
+
+      - name: Install Packages
+        run: npm ci
+
+      - name: Check for Deployment Target
+        id: check_vars
+        shell: bash
+        run: |
+          unset HAS_REGISTRY_DATA
+          if [ ! -z $DOCKER_REGISTRY -a ! -z $DOCKER_USERNAME -a ! -z $DOCKER_PASSWORD ]; then HAS_REGISTRY_DATA='true' ; fi
+          echo ::set-output name=HAS_REGISTRY_DATA::${HAS_REGISTRY_DATA}
+
+          unset HAS_UPDATE_URL
+          if [ ! -z $UPDATE_URL ]; then HAS_UPDATE_URL='true' ; fi
+          echo ::set-output name=HAS_UPDATE_URL::${HAS_UPDATE_URL}
+        env:
+          DOCKER_REGISTRY: "${{ secrets.DOCKER_REGISTRY }}"
+          DOCKER_USERNAME: "${{ secrets.DOCKER_USERNAME }}"
+          DOCKER_PASSWORD: "${{ secrets.DOCKER_PASSWORD }}"
+          UPDATE_URL: "${{ secrets.UPDATE_URL }}"
+
+      - name: Build
+        if: steps.check_vars.outputs.HAS_REGISTRY_DATA
```

Comment:
Not sure I like the idea of the build still passing if it failed to deploy because we forgot to set the SECRET.

Reply:
The main reason I did it this way was so that the actions can run in forks without needing all the secrets to be in place. (This was more useful when surge was there as well and I didn't have surge secrets but did have docker ones).

Happy to make it a hard-fail instead.

Reply:
I've made it now still attempt the build, even if it has no secrets to push it anywhere.

Reply:
Yeah, I suspect most of our forks will be creating branches to PR onto this repo anyway, if not we can always make it smarter if it becomes a problem.